src/components/tooltip/tooltip.component.ts

```diff
@@ -143,9 +142,9 @@ export default class SlTooltip extends ShoelaceElement {
     }
   };
 
-  private handleKeyDown = (event: KeyboardEvent) => {
-    // Pressing escape when the target element has focus should dismiss the tooltip
-    if (this.open && !this.disabled && event.key === 'Escape') {
+  private handleDocumentKeyDown = (event: KeyboardEvent) => {
+    // Pressing escape when a tooltip is open should dismiss it
+    if (event.key === 'Escape') {
       event.stopPropagation();
```

Comment:
Not related to the PR, but any reason we `stopPropagation()` here?

Reply:
We call this because it's listening on the document, and if you have a tooltip inside of a dialog (or similar) then the event will trigger the ancestor(s) listeners, causing them to close as well.